src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.NameGenerator.cs

```diff
@@ -57,6 +59,16 @@ private static ImmutableArray<Words> GetInterleavedPatterns(ArrayBuilder<TextSpa
                     result.Add(GetLongestBackwardSubsequence(length, breaks, baseName, pluralize));
                 }
 
+                for (var index = result.Count - 1; index >= 0; index--)
+                {
+                    // Every words-array in result corresponds to one name that will be generated.
+                    // The name should only be used if it represents a valid identifier.
+                    if (syntaxFacts.IsValidIdentifier(string.Join(string.Empty, result[index])) == false)
```

Comment:
I think I'd prefer using `CSharpSyntaxFactsService.Instance.IsValidIdentifier(...)` directly instead of passing the `Document` (or the `ISyntaxFactsService` retrieved from the `Document`) around. If another Roslyn language ever came around that wanted to also do declaration name completion, then this code would have to be shared and the `ISyntaxFactsService` used, but that doesn't seem necessary yet. Open to thoughts. :)

Reply:
That makes sense. :)